packages/synapse-react-client/src/components/ChallengeTeamWizard/MembershipRequestForm.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ChangeEvent, useImperativeHandle, useState } from 'react'
+import { Alert, Box, TextField, Typography } from '@mui/material'
+import {
+  useGetCurrentUserProfile,
+  useGetTeam,
+  useRequestToJoinTeam,
+} from '../../synapse-queries'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+import { noop } from 'lodash-es'
+
+type JoinRequestFormProps = {
+  teamId: string
+  onRequestSubmitted: () => void
+}
+
+export type MembershipRequestFormHandle = {
+  // Allow the parent component to trigger a submit of the form, so this may be embedded in an arbitrary modal.
+  submit: () => void
+}
```

Comment:
Changed to expose a `submit` button to submit the membership request, so that way we capture the logic required to make the service call in this component instead of requiring the parent to configure it.